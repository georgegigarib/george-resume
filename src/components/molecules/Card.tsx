import { CardBody } from '@/components/atoms/CardBody'
import { CardContainer } from '@/components/atoms/CardContainer'

interface Props {
  content: React.ReactNode
}

export default function Card({ content = '' }: Props) {
  return (
    <CardContainer className="inter-var">
      <CardBody className="bg-gray-50 group/card  dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl  border">
        {content}
      </CardBody>
    </CardContainer>
  )
}
