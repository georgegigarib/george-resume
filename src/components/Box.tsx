import React from 'react';
import Card from "../components/card";

interface BoxProps {
  startColumn?: number;
  startRow?: number;
  spanColumn?: number;
  spanRow?: number;
  content?: string;
}

const Box = ({
  startColumn = 1,
  startRow = 1,
  spanColumn = 2,
  spanRow = 2,
  content = 'Default Content',
}: BoxProps) => {
  const gridArea = `${startRow} / ${startColumn} / span ${spanRow} / span ${spanColumn}`;

  return (
    <div className={`rounded-3xl`} style={{ gridArea }}>
      <Card content={content}/>
    </div>
  );
};

export default Box;
