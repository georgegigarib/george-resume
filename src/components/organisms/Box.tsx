import Card from '@/components/molecules/Card'

interface BoxProps {
  startColumn?: number
  startRow?: number
  spanColumn?: number
  spanRow?: number
  content: React.ReactNode
}

const Box = ({
  startColumn = 1,
  startRow = 1,
  spanColumn = 2,
  spanRow = 2,
  content = 'Default Content'
}: BoxProps) => {
  const gridArea = `${startRow} / ${startColumn} / span ${spanRow} / span ${spanColumn}`

  return (
    <div className="rounded-xl" style={{ gridArea }}>
      <Card content={content} />
    </div>
  )
}

export default Box
