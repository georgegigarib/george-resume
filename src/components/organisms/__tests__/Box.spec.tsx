import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Box from '@/components/organisms/Box';
import Card from '@/components/molecules/Card';

describe('Box component', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <Box 
        startColumn={1}
        startRow={1}
        spanColumn={2}
        spanRow={2}
        content={<Card content="Test content" />}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
