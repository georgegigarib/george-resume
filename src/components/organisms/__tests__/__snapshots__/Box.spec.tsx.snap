// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Box component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="rounded-xl bg-transparent  shadow-3d-light dark:shadow-3d-dark transition-all hover:shadow-hover-light dark:hover:shadow-hover-dark "
    style="grid-area: 1 / 1 / span 2 / span 2;"
  >
    <div
      class="flex w-full h-full"
      style="perspective: 1000px;"
    >
      <div
        class="w-full relative transition-all duration-200 ease-linear inter-var"
        style="transform-style: preserve-3d;"
      >
        <div
          class="[transform-style:preserve-3d] [&>*]:[transform-style:preserve-3d] bg-gray-50 group/card dark:hover:shadow-2xl dark:bg-black dark:border-white/[0.2] border-black/[0.1] w-full h-full rounded-xl border"
        >
          HI
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
